src/librustc_macros/src/lib.rs

```diff
@@ -1,5 +1,6 @@
 #![feature(proc_macro_hygiene)]
 #![deny(rust_2018_idioms)]
+#![allow(default_hash_types)]
```

Comment:
Should we allow `default_hash_types` here or use `rustc_data_structures` in `rustc_macros`?

Reply:
I'm not sure, is `rustc_data_structures` using `rustc_macros` for proc macros?

Reply:
Nope `rustc_data_structures` doesn't depend on `rustc_macros`.

Reply:
Ok, adding a further dependency to `rustc_macros` seems fine to me then.

Reply:
Please, measure the build time.
Speedups from using `FxHashMap` in the proc macros are unlikely to outweigh the build parallelism lost due to the introduced dependency.

Reply:
Good point. `HashSet` is only used once in `rustc_macros` to verify, that no symbols are duplicated. 
https://github.com/rust-lang/rust/blob/03769411561f3d282bfeb0e86ca38d898f02cbdb/src/librustc_macros/src/symbols.rs#L102-L108